src/shell/shell_cmds.cpp

```diff
@@ -2506,3 +2508,151 @@ void DOS_Shell::CMD_MOVE(char* args)
+		const auto start = argsview.find_first_not_of(delimiters);
+		if (start >= argsview.size()) {
+			return {};
+		}
+
+		argsview = argsview.substr(start);
+		const auto end = argsview.find_first_of(delimiters);
+		if (end >= argsview.size()) {
+			return {};
+		}
+
+		const auto token = argsview.substr(0, end);
+		argsview         = argsview.substr(end + 1);
+		return token;
+	};
+
+	const auto variable = consume_next_token();
+	if (!variable || variable->size() != std::strlen("%") + 1 ||
+	    variable->front() != '%') {
+		SyntaxError();
+		return;
+	}
+
+	const auto in_keyword = consume_next_token();
+	if (!in_keyword || !iequals(*in_keyword, "IN")) {
+		SyntaxError();
+		return;
+	}
+
+	const auto parameters =
+	        [&argsview]() -> std::optional<std::vector<std::string>> {
+		const auto parameter_list_start = argsview.find_first_of('(');
+		if (parameter_list_start >= argsview.size()) {
+			return {};
+		}
+
+		const auto parameter_list_end = argsview.find_first_of(')');
+		if (parameter_list_end >= argsview.size() &&
+		    parameter_list_start < parameter_list_end) {
+			return {};
+		}
+
+		const auto raw_parameter_input = argsview.substr(
+		        parameter_list_start + std::strlen("("),
+		        parameter_list_end - std::strlen(")") - parameter_list_start);
+		auto raw_parameters = split(raw_parameter_input, delimiters);
+		auto expanded_parameters = std::vector<std::string>();
+
+		for (auto& str : raw_parameters) {
+			if (str.find('*') == std::string::npos &&
+			    str.find('?') == std::string::npos) {
+				expanded_parameters.emplace_back(std::move(str));
+				continue;
+			}
+
+			auto files = search_files(str);
+			expanded_parameters.insert(
+			        expanded_parameters.end(),
+			        std::make_move_iterator(files.begin()),
+			        std::make_move_iterator(files.end()));
+		}
+
+		argsview = argsview.substr(parameter_list_end + std::strlen(")"));
+		return expanded_parameters;
+	}();
+
+	if (!parameters) {
+		SyntaxError();
+		return;
+	}
+
+	const auto do_keyword = consume_next_token();
+	if (!do_keyword || !iequals(*do_keyword, "DO")) {
+		SyntaxError();
+		return;
+	}
+	
+	const auto raw_command = [&argsview]() -> std::optional<std::string_view> {
+		const auto start = argsview.find_first_not_of(delimiters);
+		if (start >= argsview.size()) {
+			return {};
+		}
+		
+		return argsview.substr(start);
+	}();
+
+	if (!raw_command)
+	{
+		SyntaxError();
+		return;
+	}
+
+	const auto format_command = [&variable, &raw_command]() -> std::string {
+		auto command  = std::string(*raw_command);
+		auto position = std::string::size_type(0);
+		while ((position = command.find(*variable, position)) !=
+		       std::string::npos) {
+			command.replace(position, variable->size(), "%s");
+			position += variable->size();
+		}
+
+		return command;
+	}();
+
+	for (const auto& parameter : *parameters) {
+		// TODO: C++20: Use std::vformat instead
```

Comment:
Yeah, I'm looking forward to C++20 as well. I think the Raspberry Pi is holding us back as only C++17 is available for it AFAIK.